jetty-ee9/jetty-ee9-nested/src/test/java/org/eclipse/jetty/ee9/nested/NcsaRequestLogTest.java

```diff
@@ -270,7 +295,7 @@ public void testBadVersion(String logType) throws Exception
 
         _connector.getResponse("METHOD /foo HTTP/9\n\n");
         String log = _entries.poll(5, TimeUnit.SECONDS);
-        assertThat(log, containsString("\"- - -\""));
+        assertThat(log, containsString("\"BAD /badMessage HTTP/1.0\""));
```

Comment:
If no URI the `HttpConnection` creates a fake one to call the `ErrorHandler`. This was causing this to be `GET /badMessage HTTP/1.0`.

 I have changed `HttpConnection` to set the Method to `BAD` so we can differentiate this from a valid request.

Reply:
Interesting, not sure about this change.
Can it be empty? or something that couldn't be sent on the protocol? (like a lowercase string with a space, eg: `<bad request>`).
Putting on my user/operations hat, seeing the hardcoded path `/badMessage` (or `HTTP/1.0`) would be a bit odd to see in the output of RequestLog too.

Reply:
I agree it is strange to have this in the RequestLog.

There is a test which does `_connector.getResponse("XXXXXXXX")` and it still shows up in RequestLog as `BAD /badMessage HTTP/1.0`.

but @gregw said it was necessary for the ErrorHandler.

maybe we could just add some flag to the request so requestLog knows to just display it as `"- - -"`?

Reply:
I'm fine with this PR being what it is.
I opened #12423 to cover the other things brought up though.